src/scala/com/github/johnynek/bazel_deps/S3URLHandler.scala

```diff
@@ -0,0 +1,422 @@
+    def AccessKeyName: String = ACCESS_KEY_ENV_VAR
+    def SecretKeyName: String = SECRET_KEY_ENV_VAR
+    
+    protected def getProp(names: String*): String = names.map{ toEnvironmentVariableName }.map{ System.getenv }.flatMap{ Option(_) }.head.trim
+  }
+  
+  private abstract class BucketSpecificCredentialsProvider(bucket: String) extends AWSCredentialsProvider {
+    def AccessKeyName: String
+    def SecretKeyName: String
+    
+    def getCredentials(): AWSCredentials = {
+      val accessKey: String = getProp(s"${AccessKeyName}.${bucket}", s"${bucket}.${AccessKeyName}")
+      val secretKey: String = getProp(s"${SecretKeyName}.${bucket}", s"${bucket}.${SecretKeyName}")
+      
+      new BasicAWSCredentials(accessKey, secretKey)
+    }
+    
+    def refresh(): Unit = {}
+    
+    // This should throw an exception if the value is missing
+    protected def getProp(names: String*): String
+  }
+
+  private abstract class RoleBasedCredentialsProvider(providerChain: AWSCredentialsProviderChain) extends AWSCredentialsProvider {
+    def RoleArnKeyNames: Seq[String]
+
+    // This should throw an exception if the value is missing
+    protected def getRoleArn(keys: String*): String
+
+    def getCredentials(): AWSCredentials = {
+      val roleArn: String = getRoleArn(RoleArnKeyNames: _*)
+
+      if (roleArn == null || roleArn == "") return null
+
+      val securityTokenService: AWSSecurityTokenService = AWSSecurityTokenServiceClient.builder().withCredentials(providerChain).build()
+
+      val roleRequest: AssumeRoleRequest = new AssumeRoleRequest()
+        .withRoleArn(roleArn)
+        .withRoleSessionName(System.currentTimeMillis.toString)
+
+      val result: AssumeRoleResult = securityTokenService.assumeRole(roleRequest)
+
+      new BasicSessionCredentials(result.getCredentials.getAccessKeyId, result.getCredentials.getSecretAccessKey, result.getCredentials.getSessionToken)
+    }
+
+    def refresh(): Unit = {}
+  }
+
+  private class RoleBasedSystemPropertiesCredentialsProvider(providerChain: AWSCredentialsProviderChain)
+      extends RoleBasedCredentialsProvider(providerChain) {
+
+    val RoleArnKeyName: String = "aws.roleArn"
+    val RoleArnKeyNames: Seq[String] = Seq(RoleArnKeyName)
+
+    protected def getRoleArn(keys: String*): String = keys.map( System.getProperty ).flatMap( Option(_) ).head.trim
+  }
+
+  private class RoleBasedEnvironmentVariableCredentialsProvider(providerChain: AWSCredentialsProviderChain)
+      extends RoleBasedCredentialsProvider(providerChain) {
+
+    val RoleArnKeyName: String = "AWS_ROLE_ARN"
+    val RoleArnKeyNames: Seq[String] = Seq("AWS_ROLE_ARN")
+
+    protected def getRoleArn(keys: String*): String = keys.map( toEnvironmentVariableName ).map( System.getenv ).flatMap( Option(_) ).head.trim
+  }
+
+  private class RoleBasedPropertiesFileCredentialsProvider(providerChain: AWSCredentialsProviderChain, fileName: String)
```

Comment:
is this class ever used? I don't see where, and it is private.

Reply:
Tool all the dependencies but forgot to remove